File: iOS/Utils/Managers/AppSettings.swift
//
//  AppSettings.swift
//  Machac
//
//  Created by Jan on 18/10/2017.
//  Copyright © 2017 Schwarja. All rights reserved.
//

import Foundation

class AppSettings {
    
    static let shared = AppSettings()
    
    let currencies = RealmManager.shared.currencies
    private(set) var referenceCurrency: Currency
    
    init() {
        if let index = currencies.index(where: { $0.isReference }) {
            referenceCurrency = currencies[index]
        }
        else {
            let czk = Currency(code: Currency.defaultCode, relationToCzk: 1, isReference: true)
            RealmManager.shared.insert(object: czk)
            referenceCurrency = czk
        }
    }
    
    func initialize() {
        
    }
}

File: Android/app/src/main/java/com/example/janschwarz1/myapplication/utils/AppSettings.java
package com.example.janschwarz1.myapplication.utils;

import com.example.janschwarz1.myapplication.models.Currency;

import io.realm.RealmResults;

/**
 * Created by janschwarz1 on 27/10/2017.
 */

public class AppSettings {

    public static AppSettings shared = new AppSettings();

    private RealmResults<Currency> currencies = RealmManager.shared.currencies();

    private Currency referenceCurrency;

    public void initialize() {
        Currency reference = RealmManager.shared.referenceCurrency();
        if (reference != null) {
            referenceCurrency = reference;
        }
        else {
            Currency czk = new Currency(Currency.defaultCode, 1.0, true);
            RealmManager.shared.write(czk);
            referenceCurrency = czk;
        }

    }

    public RealmResults<Currency> getCurrencies() {
        return currencies;
    }

    public Currency getReferenceCurrency() {
        return referenceCurrency;
    }

}
